src/ChatList/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+import { Props, ChatProps } from './props';
+import { styles } from './styles';
+
+import { ChatForm } from './ChatForm';
+import { ChatCard } from './ChatCard';
+import { ChatLoader } from './ChatLoader';
+
+import { getDateTime } from '../util/dateTime';
+
+export const ChatList: React.FC<Props> = ({
+  chats,
+  userName,
+  activeChatID = -1,
+  onChatClick = () => {},
+  onChatFormSubmit = () => {},
+  onChatLoaderVisible = () => {},
+  isLoading = false,
+  hasMoreChats = false,
+  style = {},
+}) => {
+  const loadingStyle = isLoading ? styles.loadingStyle : {};
+
+  const readLastMessage = (userName: string, chat: ChatProps) => {
+    let hasReadLastMessage = false;
+    chat.people.map((chatPerson) => {
+      if (
+        userName === chatPerson.person.username && // This is your ChatPerson instance
+        chatPerson.last_read === chat.last_message.id // and you read the last message
+      ) {
+        hasReadLastMessage = true;
+      }
+    });
```

Comment:
if you want to get fancy I think something like this would work, a little cleaner imo

```
const readLastMessage = (userName: string, chat: ChatProps) => {
    return chat.people.some(chatPerson => userName === chatPerson.person.username 
        && chatPerson.last_read === chat.last_message.id);
}

Reply:
Sexy - testing and adding if it works

TS leegend over here 😆

Reply:
LOL I was doing some reactive + webflux for some Java stuff so it was all functional interfaces